pkg/grizzly/parsing.go

```diff
@@ -0,0 +1,106 @@
+package grizzly
+
+import (
+	"bufio"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-jsonnet"
+	"github.com/grafana/tanka/pkg/jsonnet/native"
+	"github.com/grafana/tanka/pkg/kubernetes/manifest"
+	"github.com/grafana/tanka/pkg/process"
+	"gopkg.in/yaml.v3"
+)
+
+func Parse(registry Registry, opts GrizzlyOpts) (Resources, error) {
```

Comment:
I'd propose building an interface that implements a parser, similar to [the 'Loader' interface in Tanka](https://github.com/grafana/tanka/blob/master/pkg/tanka/load.go#L117). It uses a 'Detect` function to figure out which Parser to execute. All logic seems to exist here already, just have to restructure.

```go
type Parser interface {
	Parse(registry Registry, file string, opts GrizzlyOpts) (Resources, error)
}
```